### PR TITLE
Devil agent

### PR DESCRIPTION
**Pull request by lordpidey**
:cl:
add: Added Devil agent gamemode, where multiple devils are each trying to buy more souls than the next in line.
add: If you've already sold your soul, you can sell it again to a different devil.  You can even go back and forth for INFINITE POWER.
/:cl:

This adds a new gamemode.  Devil Agent.  It works much like Double Agent, except devils instead of traitors, and instead of killing, the devil simply has to control more souls than the enemy.

Wew, I'm ready for this to be reviewed.

Also, whether this is included in the rotation depends on the config settings.  By default, it will be disabled.

### CODE REVIEW

**Comment by GunHog** [comment]
I hope this pans out for you! ^_^

**Comment by jughu** [comment]
N-no more devils m-moar angels pls

File: code/game/gamemodes/devil/devil agent/devil_agent.dm [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	traitors_possible = 10 //hard limit on traitors if scaling is turned off
+	num_modifier = 6 // Six additional traitors
+	objective_count = 1
+
+	var/list/target_list = list()
+	var/list/late_joining_list = list()
+
+/datum/game_mode/devil/devil_agents/announce()
+	world << "<B>The current game mode is - Devil Agents!</B>"
+	world << "<B>There are several devils onboard the station, trying to each sell more souls than the other.</B>"
```
Comment (Jalleo) [inline code/game/gamemodes/devil/devil agent/devil_agent.dm]: Abuy buy buy. Wait did you hear i wanted to sell souls I want to buy your soul

**Comment by Bawhoppen** [comment]
gamemode code freeze

**Comment by Robustin** [comment]
Last I checked Devils were utterly shit. Why the fuck would a mode involving more of them be a good thing?

"Hey assistant, u want all-access?"
"Sure"
"Ok"
~Soul Acquired~
~Greytide Engaged~

**Comment by KorPhaeron** [comment]
Not sure our server is high enough RP for this. Devil has lots of fluffy limits you're supposed to follow/rules/lore whatever, but everyone I saw was just running around screaming "HEY BRO WANT TO SELL YOUR SOUL?"

**Comment by BananaSenor** [comment]
Joined late into such a round as lawyer. Hit the evil Captain with his contract, saw the message he is back on the team, yet he could not be revived after he died a few seconds later. Sold my own soul, deconverted myself, but kept the powers as well. Stuff seems to be broken.

**Comment by lordpidey** [comment]
@BananaSenor You keep your boons when you regain (or resell) your soul, that is intentional.  Devils are supposed to try to prevent this, it's one of the tasks they have.  However, being unrevivable after regaining your soul is a bug.

@KorPhaeron There should only be three RP limits a devil has.  Don't hurt other devils.  A random obligation, and a random forbiddance.  I'd be okay with dropping it to the don't hurt devils and one random obligation OR random forbiddance.  The Bane and Banishment traits of a devil are mechanically enforced.

**Comment by lordpidey** [comment]
Aright, I've finished the feedback I've recieved from the first test.  There were a lot of problems, and given how hectic that one round was, there will likely need to be another test to look for more problems.

**Comment by optimumtact** [comment]
So this is still in progress?

**Comment by lordpidey** [comment]
I recieved no feedback from the 2nd test, except one person was unable to move upon ascending to third stage devil (Lanky devil), and I can't figure out why, I also can't seem to reproduce the bug.  I'd like to track that bug down, but as I cannot reproduce it, I'm having a hard time.

**Comment by GeneralGoose** [comment]
sold soul = cluwnban 2 years

**Comment by lordpidey** [comment]
There was already a discussion about penalties for selling your soul.  Ultimately, it was decided that doing nothing would be best.

**Comment by GunHog** [comment]
> There was already a discussion about penalties for selling your soul. Ultimately, it was decided that doing nothing would be best.

If the Devil dies, he can steal your body, forcibly ghosting you. How about that?

**Comment by lordpidey** [comment]
@GunHog, that is a very interesting idea, but outside of the scope of this PR.  I'm not sure if I like it, as legends about selling your soul only have consequences after death.

**Comment by PKPenguin321** [comment]
Who cares about irl legends

**Pull request merged**